### PR TITLE
Delete an attribute_mapping

### LINKED ISSUE #221: Delete an attribute mapping
As attribute mappings will not be `BaseResource`s, their delete endpoint will need to be handled separately.

There should be a `delete` function in `AttributeMapping`, and a `delete_by_resource_id` in `AttributeMappingCollection`

### PR DESCRIPTION
**Pull request by charlottemoremen**
# ↪️ Fixes #221 

## ✔️ PR Todo

- [ ] Added/updated testing for this change
- [ ] Included links to related issues/PRs
- [ ] Update relevant [docs](https://github.com/Datatamer/unify-client-python/tree/master/docs) + docstrings
- [ ] Update the [CHANGELOG](https://github.com/Datatamer/unify-client-python/blob/master/CHANGELOG.md) under the current `-dev` version:
  - Add changelog entries under any that apply: **BREAKING CHANGES**, **NEW FEATURES**, **BUG FIXES**.
  - Changelog entry format: `[#<issue number>](<link to issue>) <change description>`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
 ## 0.9.0-dev
+  **BREAKING CHANGES**
+  -`AttributeMapping` now takes in `client`, `data`, and `alias` in the init
```
Comment (juliamcclellan) [inline CHANGELOG.md]: There needs to be a space between - and `

File: tamr_unify_client/project/attribute_mapping/collection.py [hunk 2]
```diff
@@ -54,4 +54,15 @@ def create(self, creation_spec):
...
         data = self.client.post(self.api_path, json=creation_spec).successful().json()
-        return AttributeMapping(data)
+        return AttributeMapping(self.client, data)
+
+    def delete_by_resource_id(self, resource_id):
+        """delete an attribute mapping using its Resource ID.
+        :param resource_id: the resource ID of the mapping to be deleted.
+        :type resource_id: str
+        :returns: :class:`requests.Response`
+        :rtype: HTTP status code
```
Comment (juliamcclellan) [inline tamr_unify_client/project/attribute_mapping/collection.py]: What's in returns and rtype should be reversed

File: tamr_unify_client/project/attribute_mapping/resource.py [hunk 3]
```diff
@@ -64,6 +66,14 @@ def resource_id(self):
         spliced = self.relative_id.split("attributeMappings/")[1]
         return spliced
 
+    def delete(self):
+        """deletes a specified attribute mapping.
+        :return: status code
+        :rtype: :class `Requests.response`
```
Comment (juliamcclellan) [inline tamr_unify_client/project/attribute_mapping/resource.py]: Response is the one that should be capitalized, not requests

File: tests/unit/test_attribute_mapping_collection.py [hunk 4]
```diff
@@ -39,23 +39,37 @@ def test_by_relative_id(self):
...
+        general_url = (
+            "http://localhost:9100/api/versioned/v1/projects/4/attributeMappings"
         )
-        map_collection = AttributeMappingCollection(
-            self.unify, "projects/4/attributeMappings"
-        )
-        test = map_collection.create(self.create_json)
-        self.assertEqual(test.input_dataset_name, self.create_json["inputDatasetName"])
-        self.assertEqual(json.loads(snoop_dict["payload"]), self.create_json)
+        responses.add(responses.GET, general_url, json=self.mappings_json)
```
Comment (juliamcclellan) [inline tests/unit/test_attribute_mapping_collection.py]: You don't actually use this get call

File: tests/unit/test_attribute_mapping.py [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
 
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
 from tamr_unify_client.project.attribute_mapping.resource import AttributeMapping
 
 
 class TestAttributeMapping(TestCase):
+    def setUp(self):
```
Comment (pcattori) [inline tests/unit/test_attribute_mapping.py]: please convert this to a pytest fixture

File: tests/unit/test_attribute_mapping.py [hunk 6]
```diff
@@ -1,11 +1,19 @@
...
+        auth = UsernamePasswordAuth("username", "password")
+        self.unify = Client(auth)
```
Comment (pcattori) [inline tests/unit/test_attribute_mapping.py]: `unify` -> `tamr`:

```suggestion
        self.tamr = Client(auth)
```

**Comment by nbateshaus** [comment]
Obsoleted by #280

**Pull request closed without merging**